src/Solution/DirectorySolution.php

```diff
@@ -29,16 +33,30 @@ class DirectorySolution implements SolutionInterface
     /**
      * @param string $directory
      * @param string $entryPoint
+     * @param array  $exclusions
      * @throws InvalidArgumentException
      */
-    public function __construct($directory, $entryPoint)
+    public function __construct($directory, $entryPoint, array $exclusions = [])
     {
         $directory  = realpath(rtrim($directory, '/'));
         $entryPoint = ltrim($entryPoint, '/');
-        
-        $files = array_values(array_diff(scandir($directory), ['..', '.']));
+
+        $dir  = new RecursiveDirectoryIterator($directory, RecursiveDirectoryIterator::SKIP_DOTS);
+        $iter = new RecursiveIteratorIterator(
+            new RecursiveCallbackFilterIterator($dir, function (\SplFileInfo $current) use ($exclusions) {
+                return !in_array($current->getBasename(), $exclusions);
+            }),
+            RecursiveIteratorIterator::SELF_FIRST
+        );
+
+        $files = [];
+        foreach ($iter as $file) {
+            if ($file->isFile()) {
+                $files[] = trim(str_replace($directory, '', $file->getPathname()), '/');
```

Comment:
maybe just do a `strlen` on directory and cut the path off at that index? It would probably be a bit faster that `str_replace` and a bit safer as the base dir string could (probably wouldn't) appear in the path elsewhere